src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,623 @@
+
+                    await proc.WaitForExitAsync();
+                    outputWaitHandle.WaitOne();
+                    errorWaitHandle.WaitOne();
+
+                    var exitCode = proc.ExitCode;
+                    return new FileCheckResult(exitCode, stdOut.ToString(), stdErr.ToString());
+                }
+            }
+            catch (Exception ex)
+            {
+                return new FileCheckResult(1, String.Empty, ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Get the method name from the method declaration.
+        /// </summary>
+        static string GetMethodName(MethodDeclarationSyntax methodDecl)
+        {
+            return
+                methodDecl.ChildTokens()
+                .OfType<SyntaxToken>()
+                .Where(x => x.IsKind(SyntaxKind.IdentifierToken)).First().ValueText;
+        }
+
+        static IEnumerable<SyntaxTrivia> GetDescendantSingleLineCommentTrivia(SyntaxNode node)
+        {
+            return
+                node
+                .DescendantTrivia()
+                .Where(x => x.IsKind(SyntaxKind.SingleLineCommentTrivia));
+        }
+
+        /// <summary>
+        /// Gather all syntactical method declarations whose body contains
+        /// FileCheck syntax.
+        /// </summary>
+        static MethodDeclarationInfo[] FindMethodsByFile(string filePath, string[] checkPrefixes)
+        {
+            var syntaxTree = CSharpSyntaxTree.ParseText(SourceText.From(File.ReadAllText(filePath)));
+            var root = syntaxTree.GetRoot();
+
+            var trivia =
+                GetDescendantSingleLineCommentTrivia(root)
+                .Where(x =>
+                {
+                    if (x.Token.Parent != null)
+                    {
+                        return !x.Token.Parent.Ancestors().Any(p => p.IsKind(SyntaxKind.MethodDeclaration) && p.Span.Contains(x.Span));
+                    }
+                    else
+                    {
+                        return true;
+                    }
+                })
+                .Where(x => ContainsCheckPrefixes(x.ToString(), checkPrefixes))
+                .ToArray();
+
+            if (trivia.Length > 0)
+            {
+                throw new Exception("FileCheck syntax not allowed outside of a method.");
+            }
+
+            return
+                root
+                .DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(x => ContainsCheckPrefixes(x.ToString(), checkPrefixes))
+                .Select(x => new MethodDeclarationInfo(x, GetMethodName(x)))
+                .ToArray();
+        }
+
+        static string? TryTransformDirective(string lineStr, string[] checkPrefixes, string syntaxDirective, string transformSuffix)
+        {
+            var index = lineStr.IndexOf(syntaxDirective);
+            if (index == -1)
+            {
+                return null;
+            }
+            else
```

Comment:
`else` isn't needed

Reply:
Yea, that is right. I think why I tend to do this sometimes is because in F#, you have to have an else if the branches return. I got some habits :)

Reply:
I think sometimes it makes sense too - if the `if`/`else` have symmetry between them then it can be artificial to unindent the `else` side.